docs/docs/events.md

```diff
@@ -181,4 +181,29 @@ class MyComponent extends Component {
 
   }
 }
-```
\ No newline at end of file
+```
+
+## Unsubscribe from the events
+Use `EventSubscription::remove` method to unsubscribe from the events
+
+```js
+class MyComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.eventSubscription = Navigation.events().bindComponent(this);
```

Comment:
> You shouldn't subscribe to event in constructor because it is not guaranteed that the component is mounted. So basically there could be situation that you bind to these events and mounting of the component has been cancelled.
> 
> So my suggestion is
> 
> 1. Move `this.eventSubscription = Navigation.events().bindComponent(this);` inside componenDidMount
> 2. move `this.eventSubscription.remove();` inside `componentWillUnmount`

Sure @henrikra.
It's just an example of EventSubscription usage.
Since wix doc wrote all examples in `constructor`, I just followed their styles.

Reply:
Good to hear. I would highly suggest you to read this whole blog post about the stuff I was talking about. But if you want to check only event listening part it is here https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#adding-event-listeners-or-subscriptions

Reply:
> Also would it makes sense to add unsub example to all code examples above? Then it is very clear to everyone how to unsub and you don't need this section down here. What do you think?

That sounds good!
Actually, there might be a call to unsubscribe the listener if we add a listener.
I totally agree with you!
I will update them right now.

Reply:
So basically what I suggest is that you update other examples as well to 2019 :D aka dont use constructor

Reply:
@wagng Thanks for PRing the docs ❤️ 
@henrikra has a valid point, I suggest you update the pr according to his comments and we can merge.

We should also note that RNN automatically unsubscribes components when they unmount, therefore unsubscribing isn't actually mandatory if you subscribed in didMount.

Reply:
@guyca Good to know it!
Yes, I am working on @henrikra 's comments now.

Reply:
💪

Reply:
@guyca You mean this line of code https://github.com/wix/react-native-navigation/blob/master/lib/src/components/ComponentWrapper.tsx#L40 ?
I don't know what is the reason behind this but what if I want to unsub earlier? And also now unsubbing is hidden so I would also add that to the documentation that you don't have to unsub because it is done by componentWillUnmount

Reply:
You can unsub earlier, we just unsubscribe as well just incase to avoid leaks.